test/integration/test_keyprotect.py

```diff
@@ -44,6 +44,21 @@ def test_disable_enable_key(self):
         resp = self.kp.get(self.key.get("id"))
         self.assertEqual(resp["state"], 1)
 
+    def test_key_ring(self):
+        # test creating key ring
+        resp = self.kp.create_key_ring(self.instance_id, "testKeyRingIdPython")
```

Comment:
If we try to delete an instance with a key ring still around, does it fail or succeed?

If we need to delete the key ring before we delete the instance, then there should be an `addCleanup` step here

Reply:
Since the key ring is already deleted in the test, adding addCleanup will cause KEY_RING_NOT_FOUND_ERR.